helper-cli/src/main/kotlin/commands/GetPackageLicensesCommand.kt

```diff
@@ -94,7 +94,7 @@ internal class GetPackageLicensesCommand : OrtHelperCommand(
 
             val detectedLicense = curatedFindings.toSpdxExpression()
 
-            val rootLicense = RootLicenseMatcher().getApplicableRootLicenseFindingsForDirectories(
+            val rootLicense = ParentLicenseMatcher().getApplicableLicenseFindingsForDirectories(
```

Comment:
I believe "Main" has the same notion as "Root" in that it can be interpreted to only mean the license in the top-level directory. More ideas:

- AncestorLicenseMatcher
- InheritedLicenseMatcher
- ApplicableLicenseMatcher
- PathLicenseMatcher

Out of these, probably `PathLicenseMatcher` is the simplest and least repetitive WRT the class method names.

Any preferences from our side, @fviernau?

Reply:
Puh, I'm having a hard time deciding this (except that I would not use Applicable because also licenses in other files can be applicable). Let's maybe do a quick poll on the preferred terminology. @mnonnenmacher @tsteenbe any preferences / ideas?

Reply:
At least @mnonnenmacher is on vacation, and I'd like to proceed with this before he's back in two weeks.

Reply:
Just one more idea: How about combining `Path` with `Main` to capture that the license is somewhat
originating from special files, not any location? `PathMainLicenseMatcher` ?

Reply:
>  to capture that the license is somewhat originating from special files

I believe that "License" in "PathLicenseMatcher" already makes clear enough that only special files, namely *license* files (which are located along the *path* to the root), are taken into account.

The term "main license" to me is more something that makes sense in the scope of a whole package, where you want to emphasize that some licenses are more important than others in that they refer to the whole package vs. only a few files in the package.

Reply:
> The term "main license" to me is more something that makes sense in the scope of a whole package, where you want to emphasize that some licenses are more important than others in that they refer to the whole package vs. only a few files in the package.

"The whole package", isn't that the only use case this matcher is about?

Reply:
From reading the test cases I thought that the use-case would be a more generic one: To check the licenses for any path, and the root directory (which implies the package license) only being a special case.

So for example, if you have no licenses files in the root, but a `dirA/LICENSE` file, and you ask for the license of `dirA/dirB`, then it would be the one of `dirA/LICENSE`. Or am I misunderstanding something?

Reply:
The algorithm normally starts at the VCS path (node) of the package. Looks inside that node for license files. 
If found returns, if not, go up. Repeat until the root.

The idea is: Normally the special files from the repository root apply. But in some cases, additionally such special files are placed in subdirs. And we treat this as override.  Note the best, but a rather common example is a
docs dir containing some Creative commons license file.

Reply:
And the use case we have implemented: Detected copyright findings which do not have a nearby detected license they can be associated with, get associated with the "main" licenses of the package. And this `PathLicenseMatcher` is determining these main licenses, starting at the VCS path of the package.